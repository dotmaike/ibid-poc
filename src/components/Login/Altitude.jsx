import React from 'react';

const Altitude = props => (
  <div className="banner">
    <div style={props.size}>ALTITUDE</div>
    <div className="subtitle">CREW INTERFACE</div>
    <div className="subtitle">Version 3.0.1</div>
  </div>
);


export default Altitude;
