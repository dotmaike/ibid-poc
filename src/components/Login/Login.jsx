import React, { Component } from 'react';
import { render } from 'react-dom';
import { Link } from 'react-router';
import Input from 'material-ui/TextField';
import Toggle from 'material-ui/Toggle';
import RButton from 'material-ui/RaisedButton';
import FButton from 'material-ui/FlatButton';
import Altitud from './Altitude.jsx';
import Paper from 'material-ui/Paper';
import Icon from '../../components/common/FAIcon.jsx';

class Login extends Component {
  render() {
    const style = {
      height: 600,
      width: 380,
      paddingTop: 80,
      display: 'inline-block',
    };

    const inputText = {
      backgroundColor: '#fff',
      height: '45px',
      marginTop: '30px',
      textIndent: '10px',
    };

    const hintText = {
      left: '8px',
      top: '43px',
      zIndex: 11,
    };

    const bannerFont = {
      fontSize: '72px',
      fontWeight: '300',
      letterSpacing: '8px'
    };

    const floatinglabel = {
      color:'#fff',
      fontSize: '15px',
      lineHeight: '0px',
      width: '350px',
      textIndent:'1px',
      textAlign: 'left',
    }

    return (
      <div className="bgLogin">
        <section className="loginWrapper">
          <div style={style} >
            <Altitud size={bannerFont} />
            <article>
              <div>
                <Input
                  inputStyle={inputText}
                  underlineShow={false}
                  hintText="Username"
                  hintStyle={hintText}
                  floatingLabelText="Please login to access your crew bidding interface"
                  floatingLabelFixed={true} floatingLabelStyle={floatinglabel}
                />
              </div>
              <div>
                <Input
                  inputStyle={inputText} underlineShow={false}
                  hintText="Password" hintStyle={hintText} type="password"
                />
              </div>
              <div className="btnWrapper">
                <Link to="/home">
                  <i className="fa fa-arrow-right" aria-hidden="true" style={{color: '#fff'}}></i>
                </Link>
              </div>
            </article>
          </div>
        </section>
        <footer className="Login-Footer">
            <div><figure>
                <img src="../../img/adopt_logo.png" />
            </figure></div>
            <div className="loginDisclaimer">
                <span>AD OPT Crew interface</span> is a registered trademark of Kronos Incorporated
            </div>
        </footer>
      </div>
    );
  }
}

export default Login;
