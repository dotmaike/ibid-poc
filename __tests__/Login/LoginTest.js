import React from 'react';
import renderer from 'react-test-renderer';
import Login from './../../src/components/Login/Login.jsx';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

/*
  Test to check if Login component exists.
*/

describe('Login (Snapshot)', () => {
    it('Login', () => {
        const component = renderer.create(<MuiThemeProvider><Login/></MuiThemeProvider>);
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
});
