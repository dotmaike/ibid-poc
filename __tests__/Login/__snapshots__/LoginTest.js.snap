exports[`Login (Snapshot) Login 1`] = `
<div
  className="bgLogin">
  <section
    className="loginWrapper">
    <div
      style={
        Object {
          "display": "inline-block",
          "height": 600,
          "paddingTop": 80,
          "width": 380,
        }
      }>
      <div
        className="banner">
        <div
          style={
            Object {
              "fontSize": "72px",
              "fontWeight": "300",
              "letterSpacing": "8px",
            }
          }>
          ALTITUDE
        </div>
        <div
          className="subtitle">
          CREW INTERFACE
        </div>
        <div
          className="subtitle">
          Version 3.0.1
        </div>
      </div>
      <article>
        <div>
          <div
            className={undefined}
            style={
              Object {
                "backgroundColor": "transparent",
                "cursor": "auto",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 16,
                "height": 72,
                "lineHeight": "24px",
                "muiPrepared": true,
                "position": "relative",
                "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": 256,
              }
            }>
            <label
              className={undefined}
              htmlFor="undefined-Username-Pleaselogintoaccessyourcrewbiddinginterface-41001"
              onTouchTap={undefined}
              style={
                Object {
                  "color": "#fff",
                  "fontSize": "15px",
                  "lineHeight": "0px",
                  "muiPrepared": true,
                  "pointerEvents": "none",
                  "position": "absolute",
                  "textAlign": "left",
                  "textIndent": "1px",
                  "top": 38,
                  "transform": "scale(0.75) translate(0, -28px)",
                  "transformOrigin": "left top",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "width": "350px",
                  "zIndex": 1,
                }
              }>
              Please login to access your crew bidding interface
            </label>
            <div
              style={
                Object {
                  "bottom": 12,
                  "color": "rgba(0, 0, 0, 0.3)",
                  "left": "8px",
                  "muiPrepared": true,
                  "opacity": 1,
                  "position": "absolute",
                  "top": "43px",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "zIndex": 11,
                }
              }>
              Username
            </div>
            <input
              disabled={false}
              id="undefined-Username-Pleaselogintoaccessyourcrewbiddinginterface-41001"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                  "appearance": "textfield",
                  "backgroundColor": "#fff",
                  "border": "none",
                  "boxSizing": "border-box",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "cursor": "inherit",
                  "font": "inherit",
                  "height": "45px",
                  "marginTop": "30px",
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "textIndent": "10px",
                  "width": "100%",
                }
              }
              type="text" />
          </div>
        </div>
        <div>
          <div
            className={undefined}
            style={
              Object {
                "backgroundColor": "transparent",
                "cursor": "auto",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 16,
                "height": 48,
                "lineHeight": "24px",
                "muiPrepared": true,
                "position": "relative",
                "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "width": 256,
              }
            }>
            <div
              style={
                Object {
                  "bottom": 12,
                  "color": "rgba(0, 0, 0, 0.3)",
                  "left": "8px",
                  "muiPrepared": true,
                  "opacity": 1,
                  "position": "absolute",
                  "top": "43px",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "zIndex": 11,
                }
              }>
              Password
            </div>
            <input
              disabled={false}
              id="undefined-Password-undefined-49269"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                  "appearance": "textfield",
                  "backgroundColor": "#fff",
                  "border": "none",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "cursor": "inherit",
                  "font": "inherit",
                  "height": "45px",
                  "marginTop": "30px",
                  "muiPrepared": true,
                  "outline": "none",
                  "padding": 0,
                  "position": "relative",
                  "textIndent": "10px",
                  "width": "100%",
                }
              }
              type="password" />
          </div>
        </div>
        <div
          className="btnWrapper">
          <a
            onClick={[Function]}
            style={Object {}}>
            <i
              aria-hidden="true"
              className="fa fa-arrow-right"
              style={
                Object {
                  "color": "#fff",
                }
              } />
          </a>
        </div>
      </article>
    </div>
  </section>
  <footer
    className="Login-Footer">
    <div>
      <figure>
        <img
          src="../../img/adopt_logo.png" />
      </figure>
    </div>
    <div
      className="loginDisclaimer">
      <span>
        AD OPT Crew interface
      </span>
       is a registered trademark of Kronos Incorporated
    </div>
  </footer>
</div>
`;
